src/scripts/main.js

```diff
@@ -1,3 +1,315 @@
+    const arr = [
+      +rows[0].children[i].innerText,
+      +rows[1].children[i].innerText,
+      +rows[2].children[i].innerText,
+      +rows[3].children[i].innerText,
+    ];
+
+    const moved = move(arr);
+
+    rows[0].children[i].innerText = moved[0] === 0 ? '' : moved[0];
+    rows[1].children[i].innerText = moved[1] === 0 ? '' : moved[1];
+    rows[2].children[i].innerText = moved[2] === 0 ? '' : moved[2];
+    rows[3].children[i].innerText = moved[3] === 0 ? '' : moved[3];
+  };
+  cellsAdder();
+};
+
+function moveDown() {
+  for (let i = 0; i < columnsQnt; i++) {
+    const arr = [
+      +rows[0].children[i].innerText,
+      +rows[1].children[i].innerText,
+      +rows[2].children[i].innerText,
+      +rows[3].children[i].innerText,
+    ];
+
+    arr.reverse();
+
+    const moved = move(arr);
+
+    moved.reverse();
+
+    rows[0].children[i].innerText = moved[0] === 0 ? '' : moved[0];
+    rows[1].children[i].innerText = moved[1] === 0 ? '' : moved[1];
+    rows[2].children[i].innerText = moved[2] === 0 ? '' : moved[2];
+    rows[3].children[i].innerText = moved[3] === 0 ? '' : moved[3];
+  };
+  cellsAdder();
+}
+
+function gameStarter() {
+  const index1 = Math.floor(Math.random() * cells.length);
+  let index2 = Math.floor(Math.random() * cells.length);
+
+  while (index1 === index2) {
+    index2 = Math.floor(Math.random() * cells.length);
+  }
+
+  const value1 = cellsCreator();
+  let value2 = cellsCreator();
+
+  while (value1 === value2) {
+    value2 = cellsCreator();
+  }
+
+  cells[index1].innerText = cellsCreator();
+  cells[index2].innerText = cellsCreator();
+
+  styleUpdater();
+};
+
+function cellsCreator() {
+  let resultCell;
+  const probability = Math.random();
+
+  if (probability < 0.1) {
+    resultCell = 4;
+  } else {
+    resultCell = 2;
+  }
+
+  return resultCell;
+}
+
+function restarter() {
+  for (const cell of cells) {
+    cell.innerHTML = '';
+    cell.classList.value = '';
+    cell.className = 'field-cell';
+  }
+  total = 0;
+  styleUpdater();
+}
+
+function cellsAdder() {
+  if (!hasEmptyCells()) {
+    return;
+  }
+
+  let wasAdded = false;
+
+  while (!wasAdded) {
+    const index = Math.floor(Math.random() * cells.length);
+
+    if (cells[index].innerText === '') {
+      cells[index].innerText = cellsCreator();
+      wasAdded = true;
+    };
+  };
+  styleUpdater();
+};
+
+function hasEmptyCells() {
+  for (const cell of cells) {
+    if (cell.innerText === '') {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+function messageUpdater() {
+  for (const cell of cells) {
+    if (cell.innerText === '2048') {
+      messageWin.classList.remove('hidden');
+    };
+  };
+};
+
+button.addEventListener('click', e => {
+  switch (e.target.innerText) {
+    case 'Start':
+      e.target.classList.value = '';
+      e.target.classList.add('button', 'restart');
+      e.target.innerText = 'Restart';
+      messageStart.classList.add('hidden');
+
+      gameStarter();
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)

Reply:
Sorry, don`t really understand what does it mean